src/core/oned/UPCAReader.ts

```diff
@@ -58,13 +55,16 @@ export default class UPCAReader extends UPCEANReader {
     return this.maybeReturnResult(this.ean13Reader.decode(image));
   }
 
-  // @Override
-  public decodeRow(rowNumber: number, row: BitArray, hints?: Map<DecodeHintType, any>): Result {
-    return this.maybeReturnResult(this.ean13Reader.decodeRow(rowNumber, row, hints));
+  public decodeRow(rowNumber: number, row: BitArray, hints?: Map<DecodeHintType, any>): Result;
+  public decodeRow(rowNumber: number, row: BitArray, startGuardRange: Int32Array, hints?: Map<DecodeHintType, any>): Result;
+  public decodeRow(rowNumber: number, row: BitArray, arg3: Int32Array | Map<DecodeHintType, any>, arg4?: Map<DecodeHintType, any>): Result {
+    const startGuardRange = arg3 instanceof Int32Array ? arg3 : UPCEANReader.findStartGuardPattern(row);
+    const hints = arg3 instanceof Map ? arg3 : arg4;
+    return this.maybeReturnResult(this.ean13Reader.decodeRow(rowNumber, row, startGuardRange, hints));
```

Comment:
Actually I saw this for overloading, I think we could improve this with some other ideas to still keep the code of each overload close to the Java version (that would be possible with multiple methods with different names but original code inside) and at the same time use only one method with the overload's name (like decodeRow) that would act as an entry point where it's implementation would choose between the implementations.